Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+
+        Parameters
+        ----------
+        args
+            One of:
+             - (str): create a Table from a file or URL
+             - (TableBase): initialize from another TableBase,
+             - (np.ndarray | scipy.sparse): a dense or sparse matrix, resulting
+               Table data type depends on the type passed, infer a domain
+                - Up to four may be passed to fit into X/Y/metas/weights respectively.
+             - (Domain, np.ndarray | scipy.sparse): Same as above, without domain inference
+             - (Domain, TableBase): convert another TableBase object to the passed domain
+             - (Domain, list): create a Table based on a list of rows.
+             - (Domain, pd.DataFrame): create a table based on an existing pandas DataFrame and Domain.
+        kwargs
+            Used exclusively for compatibility with pandas.
+
+        Returns
+        -------
+        Table | SparseTable
+        A new Table or SparseTable, depending on the arguments passed and the calling class.
+        """
+        # Needed because we have two construction paths: Table() or Table.from_X,
+        # complications arise from having a completely different signature from pandas.
+
+        # is pandas is calling this as part of its transformations, pass it through
+        all_kwargs_are_pandas = len(set(kwargs.keys()).difference(cls._KNOWN_PANDAS_KWARGS)) == 0
+
+        ##### START PANDAS SUBCLASS COMPATIBILITY SECTION #####
+        # this compatibility hack must exist because we override __new__ in a way incompatible
+        # with pandas' subclassing scheme---hacks are needed for compatibility, read on
+
+        # if we called the constructor without arguments, create empty
+        # all possible pandas kwargs will be passed to __init__, where
+        # pandas will be able to process them
+        if not args and (not kwargs or all_kwargs_are_pandas):
+            return super().__new__(cls)
+
+        # pandas can call this constructor (through Table._constructor) internally
+        # when doing transformations: its signatures are either a BlockManager or dict as the sole arg,
+        # or a ndarray as the sole arg, accompanied by at least one kwarg
+        # this serves the purpose of transforming the first arg into its matching kwarg,
+        # which will then be processed by the if clause a couple of lines above this
+        if len(args) == 1 and (isinstance(args[0], (pd.core.internals.BlockManager, dict))
+                               or (isinstance(args[0], np.ndarray) and len(kwargs) != 0 and all_kwargs_are_pandas)):
+            return cls(data=args[0], **kwargs)
+        ##### END PANDAS SUBCLASS COMPATIBILITY SECTION #####
+
+        if 'filename' in kwargs:
+            args = [kwargs.pop('filename')]
+
+        if isinstance(args[0], str):
+            if args[0].startswith('https://') or args[0].startswith('http://'):
```

Comment:
urllib, which I assume is what we are using, support FTP as well.

`str.startswith()` accepts a tuple of prefixes.